src/main/groovy/org/transmartproject/batch/tag/DeleteTagTypesTasklet.groovy

```diff
@@ -0,0 +1,27 @@
+package org.transmartproject.batch.tag
+
+import groovy.util.logging.Slf4j
+import org.springframework.batch.core.StepContribution
+import org.springframework.batch.core.scope.context.ChunkContext
+import org.springframework.batch.core.step.tasklet.Tasklet
+import org.springframework.batch.repeat.RepeatStatus
+import org.springframework.beans.factory.annotation.Autowired
+
+/**
+ * Deletes all tag types
+ */
+@Slf4j
+class DeleteTagTypesTasklet implements Tasklet {
+
+    @Autowired
+    TagTypeService tagTypeService
+
+    @Override
+    RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
+        log.info("Deleting tag types...")
+        tagTypeService.deleteAllTagsForTagTypes()
+        tagTypeService.deleteAllTagTypes()
+        RepeatStatus.FINISHED
```

Comment:
Should update the counts to indicate to the user how many items were deleted. Since you actually have two types of items, you can return the deleted tag types or some sort of total (logging at INFO the number for each type); or separate in two steps.